test/main/utilities.js

```diff
@@ -0,0 +1,43 @@
+QUnit.module( "QUnit.objectType" );
+
+function buildFunctions() {
+	var objects = [ function( ) { } ];
+
+	try {
+		var arrow = new Function( "return () => {};" )( );
+		objects.push( arrow );
+	} catch ( e ) { }
+
+	try {
+		var async = new Function( "return async function( ) { };" )( );
+		objects.push( async );
+	} catch ( e ) { }
+
+	return objects;
+}
+
+function validateObjectType( objects, expected ) {
+	QUnit.test( "should properly detect " + expected, function( assert ) {
+		assert.expect( objects.length );
+
+		for ( var i = 0; i < objects.length; i++ ) {
+			var item = objects[ i ];
+			var actual = QUnit.objectType( item );
+			assert.equal( actual, expected );
+		}
+	} );
+}
+
+validateObjectType( [ 1, Number( 1 ) ], "number" );
+validateObjectType( [ "", "foo", String( "bar" ) ], "string" );
+validateObjectType( [ null ], "null" );
+validateObjectType( [ true, false, Boolean( true ) ], "boolean" );
+validateObjectType( [ [], new Array( ) ], "array" );
```

Comment:
Awesome! Do you think I should just remove the non-primitive versions here then?